### DISCUSSION TITLE
Call to an undefined static method findMultiWithPluck in controller laravel

### DISCUSSION THREAD

**Discussion by waelsjaber** [larastan/larastan#1192] [Q&A]
I use controller  laravel

Currently I am getting:

```
Call to an undefined static method App\Models\Address::findMultiWithPluck()
```

`findMultiWithPluck()` is function custom in `MacrosServiceProvider`.

Laravel code where the issue was found:
```php

class AcademicYearController extends Controller
{

    public function index(Request $request)
    {
    
        $addresses = Address::findMultiWithPluck($request->ids);

        return response()->json($addresses);
    }
    
 }
```

is there a way to get this fixed?

**Comment by szepeviktor** [comment]
What kind of macros are in `MacrosServiceProvider`? Where are they defined? In a package?
Please provide all details.
Thanks!

**Reply by waelsjaber** [reply]
```php
class MacrosServiceProvider extends ServiceProvider
{
    public function register()
    {
      Builder::macro('findMultiWithPluck', function ($value, $key = 'name') {
            if ($value) {
                $values = explode(',', $value);

                return $this->find($values)->pluck($key)->implode(' , ');
            }

            return null;
        });
    }
}
```

**Reply by szepeviktor** [reply]
@waelsjaber Is `MacrosServiceProvider` registered in your `app/config.php` in `'providers'` array?

**Reply by waelsjaber** [reply]
@szepeviktor yes

**Reply by szepeviktor** [reply]
https://github.com/nunomaduro/larastan/pull/914
We have a test for global macros: https://github.com/nunomaduro/larastan/blob/eef292ffc5a65c15fa9ce32aec602aa68e19550d/tests/Features/Methods/Builder.php#L15-L17

**Reply by waelsjaber** [reply]
@szepeviktor 
i use `Address::findMultiWithPluck()`  static function 
if `Address::active()->findMultiWithPluck()` call scope active before findMultiWithPluck error disappears because scope active return eloquent builder.
